addons/web/static/src/views/view_hook.js

```diff
@@ -113,3 +117,68 @@ export function useBounceButton(containerRef, shouldBounce) {
         () => [containerRef.el]
     );
 }
+
+export function useExportRecords(env, context, getDefaultExportList) {
```

Comment:
we can reflect on this later, but I would consider having similar hooks for the archive and delete features. I'm not comfortable with the two new functions added in dynamic_list, especially the one taking dialogProps in arguments. I think it's too high level for the model.